instrumentation/kubernetes-client-7.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/kubernetesclient/KubernetesClientTest.java

```diff
@@ -0,0 +1,304 @@
+                                    + "/api/v1/namespaces/namespace/pods/name/proxy?path=path"),
+                            equalTo(SemanticAttributes.HTTP_METHOD, "GET"),
+                            equalTo(SemanticAttributes.USER_AGENT_ORIGINAL, TEST_USER_AGENT),
+                            equalTo(SemanticAttributes.HTTP_STATUS_CODE, 200),
+                            equalTo(SemanticAttributes.NET_PEER_NAME, "127.0.0.1"),
+                            equalTo(SemanticAttributes.NET_PEER_PORT, mockWebServer.httpPort()),
+                            satisfies(
+                                SemanticAttributes.HTTP_RESPONSE_CONTENT_LENGTH,
+                                AbstractAssert::isNotNull),
+                            equalTo(
+                                AttributeKey.stringKey("kubernetes-client.namespace"), "namespace"),
+                            equalTo(AttributeKey.stringKey("kubernetes-client.name"), "name")),
+                span ->
+                    span.hasName("callback")
+                        .hasKind(SpanKind.INTERNAL)
+                        .hasParent(trace.getSpan(0))));
+  }
+
+  @Test
+  void handleErrorsInAsynchronousCall() throws ApiException, InterruptedException {
+
+    mockWebServer.enqueue(
+        HttpResponse.of(HttpStatus.valueOf(451), MediaType.PLAIN_TEXT_UTF_8, "42"));
+
+    AtomicReference<ApiException> exceptionReference = new AtomicReference<>();
+    CountDownLatch countDownLatch = new CountDownLatch(1);
+
+    testing.runWithSpan(
+        "parent",
+        () -> {
+          coreV1Api.connectGetNamespacedPodProxyAsync(
+              "name",
+              "namespace",
+              "path",
+              new ApiCallbackTemplate() {
+                @Override
+                public void onFailure(
+                    ApiException e, int statusCode, Map<String, List<String>> responseHeaders) {
+                  exceptionReference.set(e);
+                  countDownLatch.countDown();
+                  testing.runWithSpan("callback", () -> {});
+                }
+              });
+        });
+
+    countDownLatch.await();
+
+    assertThat(exceptionReference.get()).isNotNull();
+    assertThat(mockWebServer.takeRequest().request().headers().get("traceparent")).isNotBlank();
+
+    testing.waitAndAssertSortedTraces(
+        orderByRootSpanName("parent", "get pods/proxy", "callback"),
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span -> span.hasName("parent").hasKind(SpanKind.INTERNAL).hasNoParent(),
+                span ->
+                    span.hasName("get pods/proxy")
+                        .hasKind(SpanKind.CLIENT)
+                        .hasParent(trace.getSpan(0))
+                        .hasStatus(StatusData.error())
+                        .hasEventsSatisfyingExactly(
+                            assertion -> assertApiException(exceptionReference.get(), assertion))
```

Comment:
Wouldn't
```suggestion
                        .hasException(exceptionReference.get())
```
work here as well?